amplitude_python_sdk/v2/client.py

```diff
@@ -17,10 +19,16 @@ class AmplitudeV2APIClient:  # pylint: disable=too-few-public-methods
     """
 
     def __init__(
-        self, api_key: str, api_endpoint: str = "https://api2.amplitude.com/2"
+        self,
+        api_key: str,
+        secret_key: str,
```

Comment:
This should be an optional argument, as many API methods don't require the use of the secret key. Let's make the type `Optional[str]` and set the default value to `None`.

Reply:
It might get confusing to someone who wants to use both the Batch API and Charts API but they only provide an `api_key`.  Perhaps the Charts API should have its own client?

Reply:
I think that's fine - we should think about how to best handle this going forward as many different API methods may need to have their own clients and that may get unwieldy.

Reply:
Done